docs/key-concepts.md

```diff
@@ -0,0 +1,122 @@
+
+## 2. Metrics 📊
+
+Metrics are quantitative measurements that provide insight into the **performance** and **health** of a system. Unlike logs, which provide detailed event-based information, metrics focus on numerical data points over time. Common metrics include **CPU usage**, **memory utilization**, **response times**, **throughput**, and **error rates**.
+
+Metrics play a vital role in overseeing system performance, pinpointing irregularities, and uncovering patterns that could signify either emerging problems or chances for optimization. These metrics are usually gathered and consolidated from diverse origins within the system, including applications, servers, and infrastructure components.
+
+## 3. Traces 🔍
+
+Traces are a record of the execution path of a request as it moves through a distributed system. They provide detailed information about each step taken by the request, including timestamps, service interactions, and any errors encountered along the way. 
+
+Traces help developers and operators understand how requests flow through their systems, making it easier to identify bottlenecks, diagnose problems, and optimize performance.
+
+### Instrumentation for traces
+
+Instrumentation for traces involves modifying your code to add specific instructions that capture information about how requests move through your system-specific libraries or frameworks to generate trace data as requests flow through a distributed system. This instrumentation captures details such as request origins, service interactions, and timing information at various points within the system. It typically requires adding **code snippets** or **middleware** to applications to generate and propagate trace information across different services. Examples of instrumentation libraries commonly used for tracing include OpenTelemetry, Jaeger, Zipkin, and Datadog APM.
+
+## 4. Events👁️‍🗨️
+
+Events refer to significant occurrences or actions within a system that are captured and recorded for monitoring, analysis, and processing.
+
+These events can range from routine system operations, such as application startup or shutdown, to **critical errors**, **security breaches**, or **performance anomalies**. By logging events, teams can track system behaviour, identify trends, and respond promptly to issues. Events typically contain information such as **timestamps**, **event descriptions**, **severity levels**, and **contextual data** to provide insight into what happened and when.
+
+## 5. DevSecOps🔐
+
```

Comment:
Perhaps we can add a hyperlink for a resource about `DevOps` for any folks that aren't familiar with the term `DevOps`